Source/missiles.cpp

```diff
@@ -788,196 +788,163 @@ BOOLEAN MonsterMHit(int pnum, int m, int mindam, int maxdam, int dist, int t, in
-	int v30;           // eax
-	int v32;           // [esp+Ch] [ebp-14h]
-	int arglist;       // [esp+14h] [ebp-Ch]
-	int v34;           // [esp+18h] [ebp-8h]
-	int v35;           // [esp+1Ch] [ebp-4h]
-	int dista;         // [esp+28h] [ebp+8h]
-
-	v8 = m;
-	arglist = pnum;
-	v9 = pnum;
-	v34 = m;
-	if (plr[pnum]._pHitPoints >> 6 <= 0
-	    || plr[v9]._pInvincible
-	    || plr[v9]._pSpellFlags & 1 && !missiledata[mtype].mType) {
-		return 0;
+BOOL PlayerMHit(int pnum, int m, int dist, int mind, int maxd, int mtype, BOOLEAN shift, int earflag)
+{
+	int hit, hper, tac, dam, blk, blkper, resper;
+
+	if (plr[pnum]._pHitPoints >> 6 <= 0) {
+		return FALSE;
+	}
+
+	if (plr[pnum]._pInvincible) {
+		return FALSE;
+	}
+
+	if (plr[pnum]._pSpellFlags & 1 && !missiledata[mtype].mType) {
+		return FALSE;
 	}
-	v10 = 100;
-	v32 = random(72, 100);
+
+	hit = random(72, 100);
 #ifdef _DEBUG
 	if (debug_mode_dollar_sign || debug_mode_key_inverted_v)
-		v32 = 1000;
+		hit = 1000;
 #endif
 	if (!missiledata[mtype].mType) {
-		v11 = 5;
-		v12 = plr[v9]._pIAC + plr[v9]._pIBonusAC + plr[v9]._pDexterity / 5;
-		if (v8 != -1) {
-			v11 = 2 * dist;
-			v13 = (unsigned char)monster[v8].mHit
-			    + 2 * (SLOBYTE(monster[v8].mLevel) - plr[v9]._pLevel)
+		tac = plr[pnum]._pIAC + plr[pnum]._pIBonusAC + plr[pnum]._pDexterity / 5;
+		if (m != -1) {
+			hper = monster[m].mHit
+			    + ((monster[m].mLevel - plr[pnum]._pLevel) << 1)
 			    + 30
-			    - 2 * dist;
-		LABEL_8:
-			v14 = v13 - v12;
-			goto LABEL_14;
+			    - (dist << 1) - tac;
+		} else {
+			hper = 100 - (tac >> 1) - (dist << 1);
 		}
-		v15 = v12 >> 1;
-	LABEL_12:
-		v13 = v10 - v15;
-		v12 = 2 * dist;
-		goto LABEL_8;
-	}
-	if (v8 != -1) {
-		v10 = 2 * SLOBYTE(monster[v8].mLevel) + 40;
-		v15 = 2 * plr[v9]._pLevel;
-		goto LABEL_12;
-	}
-	v14 = 40;
-LABEL_14:
-	if (v14 < 10)
-		v14 = 10;
-	if (currlevel == 14) {
-		if (v14 >= 20)
-			goto LABEL_25;
-		v14 = 20;
-	}
-	if (currlevel == 15) {
-		if (v14 >= 25)
-			goto LABEL_25;
-		v14 = 25;
-	}
-	if (currlevel == 16 && v14 < 30)
-		v14 = 30;
-LABEL_25:
-	v16 = plr[v9]._pmode;
-	if (v16 && v16 != 4 || !plr[v9]._pBlockFlag) {
-		v35 = 100;
 	} else {
-		v35 = random(73, 100);
+		if (m != -1) {
+			hper = +40 - (plr[pnum]._pLevel << 1) - (dist << 1) + (monster[m].mLevel << 1);
+		} else {
+			hper = 40;
+		}
+	}
+
+	if (hper < 10)
+		hper = 10;
+	if (currlevel == 14 && hper < 20) {
+		hper = 20;
+	}
+	if (currlevel == 15 && hper < 25) {
+		hper = 25;
+	}
+	if (currlevel == 16 && hper < 30) {
+		hper = 30;
+	}
+
+	if (!((plr[pnum]._pmode && plr[pnum]._pmode != PM_ATTACK) || !plr[pnum]._pBlockFlag)) {
+		blk = random(73, 100);
+	} else {
+		blk = 100;
 	}
-	if ((_BYTE)shift == 1)
-		v35 = 100;
+
+	if (shift == TRUE)
+		blk = 100;
 	if (mtype == MIS_ACIDPUD)
-		v35 = 100;
-	if (v8 == -1)
-		v17 = plr[v9]._pBaseToBlk;
+		blk = 100;
+	if (m != -1)
+		blkper = plr[pnum]._pBaseToBlk + (plr[pnum]._pLevel << 1) - (monster[m].mLevel << 1) + plr[pnum]._pDexterity;

```

Comment:
blkper/blk is swapped compared to MonsterMHit, I think i agree with you that this one is the right one and the one in MonsterMHit should be swapped.